### PR TITLE
The Lo-Cal Localization Nation

### PR DESCRIPTION
**Pull request by sarahsturgeon**
## TODO:
- [x] Test

## Notes
This is mostly a framework for localization. It allows individual word translations, and long sentences with formatting, so it pretty much covers all of the bases. There's an example for all use cases in the changes.

Down the line we may want to look into allowing clients to set their own language, but that'll get kind of complicated. I'm thinking that the server should set the global language, as it's likely rare that someone who's unable to speak the server's language will still play on that server (read: more issues arise from not speaking the language than not understanding the gamemode)

I left the commands alone, it'll be a bit of a hassle to change the commands based on language and I imagine some grammatical issues could pop up, too, but I'm happy to add it if you think it's necessary.

### CODE REVIEW

**Comment by samuelWilliams99** [comment]
This man has the most professional background out of all of us, I'm in *awe* of his talent

File: gamemode/localization.lua [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    en = "Canot give less than 1 item, what are you expecting to happen?",
+    fr = "Je suis une baguette?"
+}
+
+L.itemTypeDoesNotExist = {
+    en = "Item type %s does not exist",
+    fr = "Je suis une %s"
+}
+
+
```
Comment (samuelWilliams99) [inline gamemode/localization.lua:113]: ew

**Review by samuelWilliams99 — CHANGES_REQUESTED** [review]
We might need some nice way to automatically translate item displaynames, not entirely sure how that would look though

File: gamemode/localization.lua (hunk 1 above)
Comment (PLally) [inline gamemode/localization.lua:47]:
```suggestion
```

**Comment by samuelWilliams99** [comment]
Since the config is exposed to the end user, should we validate the language they input? The errors will likely be pretty confusing if they enter a language we don't support. We can put something meaningful in console and fallback to english

**Comment by sarahsturgeon** [comment]
Wow peri just pushing to my pr instead of making comments 😒

**Comment by samuelWilliams99** [comment]
Teamwork!

**Review by plally — APPROVED** [review]

File: gamemode/localization.lua [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+L.unknownCommand = {
+    en = "Command \"%s\" does not exist",
+    fr = "<UNKNOWN>"
+}
+
+L.commandsInConsole = {
+    en = "Look in console for a list of commands.",
+    fr = "<UNKNOWN>"
+}
```
Comment (overlordbleck) [inline gamemode/localization.lua:117]: Newline at the end pls
```suggestion
}

```

File: gamemode/shared.lua [hunk 3]
```diff
@@ -5,6 +5,8 @@ GM.Website = "https://github.com/cfc-servers/raft_v_raft"
 RVR = RVR or {}
 
 AddCSLuaFile( "load_config.lua" )
+AddCSLuaFile( "localization.lua" )
```
Comment (overlordbleck) [inline gamemode/shared.lua:8]: nitpicking, but I'm pretty sure this should be `Localisation` not `Localization`

File: gamemode/shared.lua [hunk 4]
```diff
@@ -5,6 +5,8 @@ GM.Website = "https://github.com/cfc-servers/raft_v_raft"
...
 AddCSLuaFile( "load_modules.lua" )
 include( "load_config.lua" )
+include( "localization.lua" )
```
Comment (overlordbleck) [inline gamemode/shared.lua:11]: Same here

**Review by overlordbleck — APPROVED** [review]

File: gamemode/shared.lua (hunk 4 above)
Comment (overlordbleck) [reply gamemode/shared.lua:11]: Looking back I'm seeing a lot of places where you use `Localization` :octocat:

File: gamemode/shared.lua (hunk 3 above)
Comment (plally) [reply gamemode/shared.lua:8]: I think both are valid spellings

**Review by plally — APPROVED** [review]

File: gamemode/shared.lua (hunk 3 above)
Comment (sarahsturgeon) [reply gamemode/shared.lua:8]: `Localization` is typically the US spelling (`Localisation` being the UK spelling); I specifically went with it 'cause the englishman

Comment (samuelWilliams99) [reply gamemode/shared.lua:8]: I'm very upset

**Pull request closed without merging**